syntax = "proto3";

package request_proto;

import "Compliance.proto";

service Regulator {
    rpc GetComplianceProof (stream ProofRequest) returns (stream ProofResponse) {}
}

message ProofRequest {
    uint64 nonce = 1; // A unique number for this request message
    string peer_signature = 2;
    TransactionInfo tx_info = 3;
}

message ProofResponse {
    uint64 nonce = 1; // A unique number for this response message
    CProof c_proof = 2;
}